pulp_file/tests/functional/api/test_crud_remotes.py

```diff
@@ -128,6 +128,82 @@ def test_all(self):
             api.Client(config.get_config()).post(FILE_REMOTE_PATH, body)
 
 
+class RemoteDownloadPolicyTestCase(unittest.TestCase):
+    """Verify download policy behavior for valid and invalid values.
+
+    In Pulp 3, there are are different download policies.
+
+    This test targets the following testing scenarios:
+
+    1. Creating a remote without a download policy.
+       Verify the creation is successful and immediate it is policy applied.
+    2. Change the remote policy from default.
+       Verify the change is successful.
+    3. Attempt to change the remote policy to an invalid string.
+       Verify an HTTPError is given for the invalid policy as well
+       as the policy remaining unchanged.
+
+    For more information on the remote policies, see the Pulp3
+    API on an installed server:
+
+    * /pulp/api/v3/docs/#operation`
+
+    This test targets the following issues:
+
+    * `Pulp #4420 <https://pulp.plan.io/issues/4420>`_
+    * `Pulp #3763 <https://pulp.plan.io/issues/3763>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.remote = {}
+        cls.body = _gen_verbose_remote()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean class-wide variable."""
+        cls.client.delete(cls.remote['_href'])
+
+    def test_01_no_defined_policy(self):
+        """Verify the default policy `immediate`.
+
+        When no policy is defined, the default policy of `immediate`
+        is applied.
+        """
+        del self.body['policy']
+        self.remote.update(self.client.post(FILE_REMOTE_PATH, self.body))
+        self.assertEqual(self.remote['policy'], 'immediate', self.remote)
+
+    def test_02_change_policy(self):
+        """Verify ability to change policy to value other than the default.
+
+        Update the remote policy to a valid value other than `immedaite`
+        and verify the new set value.
+        """
+        remove_policy = DOWNLOAD_POLICIES.copy()
+        remove_policy.remove('immediate')
+        changed_policy = choice(remove_policy)
```

Comment:
@kersommoura talked with bruno, and he had the suggestion of removing L186-188 with:
```suggestion
        self.client.patch(self.remote['_href'], {'policy': choice([item for item in DOWNLOAD_POLICIES if item != 'immediate'])})
```

As long as in `constants.py` I changed the `DOWNLOAD_POLICIES` from a `list` to a `tuple`.

The only other code to use `DOWNLOAD_POLICIES` in this project was L223. Therefore, the change should not be an issue.

Let me know if you have any comments.

Reply:
That is a better way to handle this.

```suggestion
changed_policy = choice(
      [item for item in DOWNLOAD_POLICIES if item != 'immediate']
)
self.client.patch(self.remote['_href'], {'policy': changed_policy })
```

We will need to know ``changed_policy`` in order to make the assertion.

Reply:
hehe... that is JUST what I got done implementing and formatting (reading PEP-8 [0]).

Get on vacation, @kersommoura ! This isn't critical! ;)

Thank you!

[0] - https://www.python.org/dev/peps/pep-0008/#multiline-if-statements

Reply:
@kersommoura , do you then suggest we change the import of `DOWNLOAD_POLICIES` in `test_crud_remotes.py` to `pulp_smash` and removal of this line from `constants.py`? 

If so, I can open up a PR after this to address that change. 

Still, not sure why even there it was added as a list, but that `horse is out of the barn`. 

Thanks for your time, and enjoy your vacation!